src/main/java/touro/chess/BishopPiece.java

```diff
@@ -0,0 +1,49 @@
+package touro.chess;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BishopPiece extends AbstractPiece
+{
+    public BishopPiece(Location location, PieceColor color)
+    {
+        super(location, color);
+    }
+
+    @Override
+    public List<Move> getMoves()
```

Comment:
Can you think of a way to do this in 2 for loops instead of 4 for loops? Or even one for loop?

Reply:
I don't see how I could do it with less than 4 for loops since the rows and columns increment and decrement differently for each direction. Is there a different way I should be approaching it that would be more efficient?